src/util/antoraSupport.ts

```diff
@@ -0,0 +1,36 @@
+import vscode from 'vscode'
+import fs from 'fs'
+import yaml from 'js-yaml'
+import path from 'path'
+
+export default class AntoraSupport {
+  public async isDetected (): Promise<boolean> {
+    const antoraYmlFiles = await vscode.workspace.findFiles('**/antora.yml', '/node_modules/', 10)
+    return antoraYmlFiles.length > 0
+  }
+
+  public async getAntoraYmlPath (textDocument: vscode.TextDocument): Promise<any> {
+    const pathToAsciidocFile = textDocument.uri.fsPath
+    const antoraYmlFiles = await vscode.workspace.findFiles('**/antora.yml', '/node_modules/', 100)
+    // check if antora.yml file matches with textDocument content source root
+    try {
+      for (const antoraYmlFile of antoraYmlFiles) {
+        if (pathToAsciidocFile.startsWith(path.dirname(antoraYmlFile.path))) {
+          return antoraYmlFile.path
+        }
+      }
+    } catch (err) {
+      console.log('Antora file not applicable to current asciidoc document, ' + err.toString())
+    }
+  }
+
+  public async getAttributes (textDocument: vscode.TextDocument): Promise<{ [key: string]: string }> {
+    try {
+      const doc = yaml.load(fs.readFileSync(await this.getAntoraYmlPath(textDocument), 'utf8'))
+      return doc.asciidoc.attributes
+    } catch (err) {
+      console.log('Unable to parse /path/to/docs/antora.yml, cause:' + err.toString())
```

Comment:
`/path/to/docs/antora.yml` should be the *actual* path to the `antora.yml` file.